app/views/publish/courses/student_visa_sponsorship/edit.html.erb

```diff
@@ -9,14 +9,21 @@
           course.course_code
         )
       ) %>
-  <% elsif @provider.accredited_provider? %>
+  <% elsif @provider.accredited_provider? && params[:previous_tda_course] != 'true' %>
     <%= govuk_back_link_to(
       apprenticeship_publish_provider_recruitment_cycle_course_path(
           course.provider_code,
           course.recruitment_cycle_year,
           course.course_code
         )
     ) %>
+  <% elsif params[:previous_tda_course] == 'true' %>
+    <%= govuk_back_link_to(full_part_time_publish_provider_recruitment_cycle_course_path(
+              course.provider_code,
+              course.recruitment_cycle_year,
+              course.course_code,
+              previous_tda_course: true
+            )) %>
```

Comment:
This erb file is getting out of hand with logic. We've got 33 lines of an if / else statement here, making this very difficult to read. The erb files should just place the elements on the page and logic should be very minimal.

There are a few different ways to do it -- perhaps in the decorator, the form or the controller. 

But it should look more like:
```
<% content_for :before_content do %>
  <%= govuk_back_link_to(back_link)
  # or   <%= govuk_back_link_to(course.back_link(:student_visa_sponsorship)) 
<% end %>
```

The logic itself is rather difficult to read. If you are checking something is not true and then checking that it is true in the next condition, this will be difficult for future you / other developers to comprehend later without drawing a complex venn diagram.

You should get the same results with 
```
def back_link
  if params[:back_to_basic_details].present?
    ...
  elsif params[:previous_tda_course] == 'true'
    ...
  elsif @provider.accredited_provider?
    ...
  else 
    ...
  end
end
```

Reply:
Discussed offline, and decided to go with: https://github.com/DFE-Digital/publish-teacher-training/pull/4284/commits/3acddbbd14034e43ec8b9a690907e634eefd2e3d